src/main/java/com/rackspace/ceres/app/services/QueryService.java

```diff
@@ -101,11 +120,29 @@ public QueryService(ReactiveCqlTemplate cqlTemplate,
         .checkpoint();
   }
 
-  public Flux<QueryResult> queryDownsampled(String tenant, String metricName, Aggregator aggregator,
+  public Flux<QueryResult> queryDownsampled(String tenant, String metricKey, Aggregator aggregator,
       Duration granularity, Map<String, String> queryTags,
       Instant start, Instant end) {
+    return getMetricGroupFlux(metricKey)
+        .flatMap(
+            metricName -> {
+              return getQueryDownsampled(tenant, metricName, aggregator, granularity, queryTags, start, end);
+            }
+        )
+        .switchIfEmpty(getQueryDownsampled(tenant, metricKey, aggregator, granularity, queryTags, start, end))
+        .checkpoint();
+  }
+
+  private Flux<String> getMetricGroupFlux(String metricKey) {
+    return reactiveRedisTemplate.opsForSet().members(metricKey)
```

Comment:
There is an implicit functionality here that is not clear. If metricKey is a metricName it will reply with a single metricName and if a group it will reply with one or more names in that group. What happens if there is a name clash so a metricName is also the name of a group? We shouldn't implement like this IMHO. I think working with 2 separate things group and name is clearer and no risk of naming collision.

Reply:
I agree with your point. ideally we should go for implementing it with metricName and metricGroup individually. let's wait for @harageth 's inputs on this?